sdonboarding.client/src/components/UITemplate/AddDataTemplate.jsx

```diff
@@ -0,0 +1,304 @@
+    const getOtherObjectsData = useGetOtherObjectsData();
+    const handleAddCancel = useAddCancel(type);
+
+    /* Only relevant for Sales functionality*/
+    const [localCustomers, setLocalCustomers] = useState(null);
+    const [localProducts, setLocalProducts] = useState(null);
+    const [localStores, setLocalStores] = useState(null);
+
+    /* Fetches all the objects that we will need for Sales Add dropdowns*/
+    const fetchOthersData = useCallback(async () => {
+        try {
+            if (type === ObjectTypes.Sale) {
+                const { customers, products, stores } = await getOtherObjectsData();
+                setLocalCustomers(customers);
+                setLocalProducts(products);
+                setLocalStores(stores);
+
+                //initialize localObject with proper values so that if there is no change triggered(onChange)
+                //we have the default first value to create the sale with
+                /*setLocalObject((prevState) => ({
+                    ...prevState,
+                    customerId: customers[0].id,
+                    customer: customers[0].name,
+                    productId: products[0].id,
+                    product: products[0].name,
+                    storeId: stores[0].id,
+                    store: stores[0].name,
+                }))*/
+            }
+            
+        } catch (error) {
+            console.error('Error fetching data:', error);
+            dispatch(setError('Error fetching data for Sales:', error));
+        }
+    }, [dispatch]);
+
+    useEffect(() => {
+       
+
+        fetchOthersData();
+    }, [fetchOthersData]);
+
+    const handleInputChange = (e) => {
+        const { name, value } = e.target;
+        e.target.setCustomValidity('');
+        setLocalObject((prevState) => ({
+            ...prevState,
+            [name]: value,
+        }));
+    };
+
+    /*handle the product active dropdown value*/
+    const handleActiveChange = (e) => {
+        setLocalObject((prevState) => ({
+            ...prevState,
+            active: e.target.value === 'true'
+        }));
+    };
+
+    /*List Change handler, it will update localObject with ID and name as we select it from the dropdown*/
+    const handleListChange = (id, name, type) => {
+        const selectedType = type;
+        if (selectedType === 'Customer') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                customerId: id,
+                customer: name,
+            }))
+        } else if (selectedType === 'Product') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                productId: id,
+                product: name,
+            }))
+        } else if (selectedType === 'Store') {
+            setLocalObject((prevState) => ({
+                ...prevState,
+                storeId: id,
+                store: name,
+            }))
+        }
```

Comment:
While this is working, you can probably set up or update this by:
1. Use an enum to determine the type
2. Use switch instead of if. In this case, what will happen if let's say the **selectedType** mistakenly used something like an incorrect value